crates/shared/src/zeroex_api.rs

```diff
@@ -75,6 +81,170 @@ impl SwapQuery {
+    /// If left unspecified, anyone can fill the order.
+    pub taker: Option<H160>,
+    /// Allows the maker to enforce that the order flow through some
+    /// additional logic before it can be filled (e.g., a KYC whitelist).
+    pub sender: Option<H160>,
+    /// Address of the contract where the transaction should be sent,
+    /// usually this is the 0x exchange proxy contract.
+    pub verifying_contract: Option<H160>,
+}
+
+impl OrdersQuery {
+    /// Encodes the orders query as a url with parameters.
+    fn format_url(&self, base_url: &Url) -> Url {
+        let mut url = base_url
+            .join("/orderbook/v1/orders")
+            .expect("unexpectedly invalid URL segment");
+
+        if let Some(taker) = self.taker {
+            url.query_pairs_mut().append_pair("taker", &addr2str(taker));
+        }
+        if let Some(sender) = self.sender {
+            url.query_pairs_mut()
+                .append_pair("sender", &addr2str(sender));
+        }
+        if let Some(verifying_contract) = self.verifying_contract {
+            url.query_pairs_mut()
+                .append_pair("verifyingContract", &addr2str(verifying_contract));
+        }
+
+        url
+    }
+}
+
+impl Default for OrdersQuery {
+    fn default() -> Self {
+        Self {
+            taker: Some(H160::zero()),
+            sender: Some(H160::zero()),
+            verifying_contract: Some(DefaultZeroExApi::DEFAULT_VERIFICATION_CONTRACT),
+        }
+    }
+}
+
+#[derive(Debug, Derivative, Clone, Deserialize, PartialEq)]
+#[derivative(Default)]
+#[serde(rename_all = "camelCase")]
+pub struct OrderMetaData {
+    #[derivative(Default(value = "chrono::MIN_DATETIME"))]
+    pub created_at: DateTime<Utc>,
+    pub order_hash: Bytes,
+    pub remaining_fillable_taker_amount: U256,
+}
+
+fn deserialize_epoch_timestamp<'de, D>(deserializer: D) -> Result<DateTime<Utc>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let epoch: &str = Deserialize::deserialize(deserializer)?;
+    let naive = NaiveDateTime::from_timestamp(epoch.parse().map_err(serde::de::Error::custom)?, 0);
+    Ok(DateTime::from_utc(naive, Utc))
+}
+
+#[derive(Debug, Derivative, Clone, Deserialize, PartialEq)]
+#[derivative(Default)]
+#[serde(rename_all = "camelCase")]
+pub struct ZeroExSignature {
+    r: H256,
+    s: H256,
+    v: u8,
+    // for testing purposes default initialized as valid
+    #[derivative(Default(value = "2"))]
+    signature_type: u8,
+}
+
+impl TryFrom<ZeroExSignature> for Signature {
+    type Error = anyhow::Error;
+
+    fn try_from(signature: ZeroExSignature) -> Result<Self, Self::Error> {
+        let ecdsa = EcdsaSignature {
+            r: signature.r,
+            s: signature.s,
+            v: signature.v,
+        };
+
+        match signature.signature_type {
+            2 => Ok(Signature::Eip712(ecdsa)),
+            3 => Ok(Signature::EthSign(ecdsa)),
+            _ => Err(anyhow::anyhow!("unsupported signature type")),
+        }
+    }
+}
```

Comment:
Not sure we need this conversion, since 0x signatures are different from ours (it might also exclude orders signed in different ways that we could support). In fact these signatures get encoded in a different way (they will be added to calldata for an interaction and not part of the settlement trades), so I think its fine just keeping `ZeroExSignature`s for these orders and not converting them into `Signature`.